pkg/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -28,7 +28,7 @@ func errorNotAMap(component string) error {
 	return fmt.Errorf("%s property in the configuration doesn't contain valid %s", component, component)
 }
 
-// ConfigToPromConfig converts the incoming configuration object into a the Prometheus receiver config.
+// ConfigToPromConfig converts the incoming configuration object into the Prometheus receiver config.
 func ConfigToPromConfig(cfg string) (map[interface{}]interface{}, error) {
```

Comment:
I think I'd rather properly fix this check, as it's currently incorrect. Right now, it requires that the `config` property exists, but in reality we should check if one of the following is true:
* `config` exists
* `target_allocator` exists and target allocator is enabled on the resource
* target allocator is enabled on the resource and the feature flag is enabled

Then this is really a fix, not a breaking change, and the actual breaking change will happen when the flag is enabled by default. How's that sound?

Reply:
yes, technically the comment on the function doesn't match what the code is actually doing, but regardless anyone potentially using this function and expecting it to return the prometheus config blob would be broken by this change. I think i'm okay to call this a 'fix', as long as we call that out in the changelog's subtext – in case someone is using this for some reason they should be able to see the difference in the release notes.